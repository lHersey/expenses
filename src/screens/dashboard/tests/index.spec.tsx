import { render } from '@testing-library/react-native';
import React, { ComponentProps, FC } from 'react';
import Providers from 'shared/providers';

import BaseDashboard from '../index';

const Dashboard: FC<ComponentProps<typeof BaseDashboard>> = props => {
  return (
    <Providers>
      <BaseDashboard {...props} />
    </Providers>
  );
};

describe('<Dashboard />', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
      expect.hasAssertions();
      const rendered = render(<Dashboard />).toJSON();

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('unit', () => {
    it('should render the text', () => {
      expect.hasAssertions();
      const { queryByText } = render(<Dashboard />);

      expect(queryByText('Hello Dashboard')).toBeTruthy();
    });
  });
});
