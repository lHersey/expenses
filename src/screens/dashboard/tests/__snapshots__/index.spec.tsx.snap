// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> snapshots should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNCSafeAreaView
    style={
      Array [
        Object {
          "backgroundColor": "#f2f5f7",
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingLeft": 30,
            "paddingTop": 20,
          },
        ]
      }
    >
      <Text />
      <Text
        style={
          Array [
            Object {
              "color": "#5e5e5e",
              "fontFamily": "Inter_500",
              "fontSize": 16,
            },
            Object {
              "color": "#5e5e5e",
              "fontFamily": "Inter_600",
              "fontSize": 20,
              "marginLeft": 20,
            },
          ]
        }
      >
        Welcome to Edify
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingLeft": 30,
            "paddingRight": 30,
            "paddingTop": 30,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "visible",
            },
          ]
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4280987875,
                4292112360,
                4294874740,
              ],
              "endPoint": Array [
                1,
                0,
              ],
              "locations": undefined,
              "startPoint": Array [
                0,
                0,
              ],
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "#FFF",
                "borderRadius": 15,
                "display": "flex",
                "flexGrow": 1,
                "overflow": "hidden",
                "paddingBottom": 15,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 15,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "marginTop": 10,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#5e5e5e",
                    "fontFamily": "Inter_500",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#e3e3e3",
                  },
                ]
              }
            >
              Total Balance
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "marginTop": 10,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#5e5e5e",
                    "fontFamily": "Inter_500",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#e3e3e3",
                    "fontFamily": "Inter_600",
                    "fontSize": 36,
                  },
                ]
              }
            >
              $
              0
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginTop": 20,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0.15)",
                    "borderRadius": 10,
                    "display": "flex",
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "marginBottom": 10,
                    "marginLeft": 10,
                    "marginRight": 10,
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                ]
              }
            >
              <Text />
              <View
                style={
                  Array [
                    Object {
                      "marginLeft": 10,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#5e5e5e",
                        "fontFamily": "Inter_500",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#e3e3e3",
                        "fontFamily": "Inter_700",
                        "fontSize": 14,
                      },
                    ]
                  }
                >
                  Income
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#5e5e5e",
                        "fontFamily": "Inter_500",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#e3e3e3",
                        "fontSize": 14,
                      },
                    ]
                  }
                >
                  $
                  0
                  .00
                </Text>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0.15)",
                    "borderRadius": 10,
                    "display": "flex",
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "marginBottom": 10,
                    "marginLeft": 10,
                    "marginRight": 10,
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                ]
              }
            >
              <Text />
              <View
                style={
                  Array [
                    Object {
                      "marginLeft": 10,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#5e5e5e",
                        "fontFamily": "Inter_500",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#e3e3e3",
                        "fontFamily": "Inter_700",
                        "fontSize": 14,
                      },
                    ]
                  }
                >
                  Expenses
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#5e5e5e",
                        "fontFamily": "Inter_500",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#e3e3e3",
                        "fontSize": 14,
                      },
                    ]
                  }
                >
                  $
                  0
                  .00
                </Text>
              </View>
            </View>
          </View>
        </ViewManagerAdapter_ExpoLinearGradient>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginTop": 40,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 20,
              "paddingBottom": 0,
              "paddingLeft": 30,
              "paddingRight": 30,
              "paddingTop": 0,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#5e5e5e",
                "fontFamily": "Inter_500",
                "fontSize": 16,
              },
              Object {
                "fontFamily": "Inter_700",
                "fontSize": 20,
              },
            ]
          }
        >
          Transactions
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#5e5e5e",
                "fontFamily": "Inter_500",
                "fontSize": 16,
              },
              Object {
                "color": "#b0b0b0",
                "fontSize": 15,
              },
            ]
          }
        >
          View All
        </Text>
      </View>
      <RCTScrollView
        ItemSeparatorComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "displayName": "Styled(View)",
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
  margin: 10px 0;
",
              ],
            },
            "render": [Function],
            "shouldForwardProp": undefined,
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
          }
        }
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Array [
            Object {
              "paddingBottom": 0,
              "paddingLeft": 30,
              "paddingRight": 30,
              "paddingTop": 0,
            },
          ]
        }
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </RNCSafeAreaView>
</RNCSafeAreaProvider>
`;
